src/hotspot/share/gc/g1/g1ParScanThreadState.inline.hpp

```diff
@@ -43,7 +43,7 @@ template <class T> void G1ParScanThreadState::do_oop_evac(T* p) {
   if (in_cset_state.is_in_cset()) {
     markOop m = obj->mark_raw();
     if (m->is_marked()) {
-      obj = (oop) m->decode_pointer();
+      obj = UseAltGCForwarding ? obj->forwardee(m) : (oop) m->decode_pointer();
```

Comment:
do we have performance numbers for this change?

Reply:
I didn't see noticeable performance impact in specjbb2015

Reply:
once the new flow is stable in real work loads. I guess we can remove the condition.